libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1164,6 +1165,19 @@ blk_hash_t PbftManager::calculateOrderHash(std::vector<DagBlock> const &dag_bloc
   return dev::sha3(order_stream.out());
 }
 
+std::optional<blk_hash_t> findClosestAnchor(const std::vector<blk_hash_t> &ghost, std::vector<blk_hash_t> &dag_order,
+                                            uint32_t included) {
+  for (uint32_t i = included; i > 0; i--) {
+    if (std::count(ghost.begin(), ghost.end(), dag_order[i - 1])) {
+      if (dag_order.size() > i) {
+        dag_order.erase(dag_order.begin() + i, dag_order.end());
+      }
```

Comment:
I think we don't need remove dag blocks here, because we need call getDagBlockOrder later